gateway/conf/nginx.conf.liquid

```diff
@@ -122,7 +122,18 @@ http {
 
     access_log {{ access_log_file | default: "/dev/stdout" }} time;
 
-    listen {{ port.apicast | default: 8080 }};
+    {%- assign http_port = port.apicast | default: 8080 %}
+    {%- assign https_port = env.APICAST_HTTPS_PORT %}
+
+    {% if http_port != https_port -%}
```

Comment:
I'd add a new test case in `t/listen-https.t` just to be sure.

Reply:
So like the variable not set at all? That works - normal blackbox tests run with https_port not set.

The issue was that `{% if env.somevar %}` would be true even if it is empty.
This works just fine because `http_port` always has value.

Reply:
Right :+1: